### PR TITLE
Return a sized Enumerator from Hash#transform_values{!}

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

File: activesupport/test/core_ext/hash/transform_keys_test.rb [hunk 1]
```diff
@@ -21,12 +21,14 @@ class TransformKeysTest < ActiveSupport::TestCase
   test "transform_keys returns an Enumerator if no block is given" do
     original = { a: 'a', b: 'b' }
     enumerator = original.transform_keys
+    assert_nil enumerator.size
```
Comment (matthewd) [inline activesupport/test/core_ext/hash/transform_keys_test.rb]: Can't we teach `transform_keys` to do this too?

Comment (kachick) [inline activesupport/test/core_ext/hash/transform_keys_test.rb]: For below case. I believe `transform_keys{!}` can't settlement own size :smiley: 
```ruby
origin = {a: 1, A: 2}
origin.transform_keys(&:downcase) #=> {:a=>2}
```

Comment (matthewd) [inline activesupport/test/core_ext/hash/transform_keys_test.rb]: I think `Enumerator#size` is the number of times it'll *yield*, not the size of whatever might get returned. So, `origin.transform_keys.size #=> 2`.

Comment (kachick) [inline activesupport/test/core_ext/hash/transform_keys_test.rb]: Oops... You're right! I'll update this patch.

**Comment by kachick** [comment]
:bow:

**Pull request merged**